### PR TITLE
Fix parenthesized expressions in braceless `if` conditions

### LINKED ISSUE #263: Braceless `if` with args in parens and field expressions
### Commit of tree-sitter-scala you tested this on

374709c

### A code sample showing the error

```
object A:
  if (a) && b.c then
    ()
```

### Show the error node

```
(compilation_unit [0, 0] - [3, 0]
  (object_definition [0, 0] - [3, 0]
    name: (identifier [0, 7] - [0, 8])
    body: (template_body [0, 8] - [3, 0]
      (if_expression [1, 2] - [1, 20]
        condition: (parenthesized_expression [1, 5] - [1, 8]
          (identifier [1, 6] - [1, 7]))
        consequence: (postfix_expression [1, 9] - [1, 20]
          (field_expression [1, 9] - [1, 15]
            value: (operator_identifier [1, 9] - [1, 11])
            (ERROR [1, 12] - [1, 13])
            field: (identifier [1, 14] - [1, 15]))
          (identifier [1, 16] - [1, 20])))
      (unit [2, 4] - [2, 6]))))
```

### What do you expect the tree to look like

_No response_

### Where are you experiencing this error?

[Namer.scala](https://github.com/lampepfl/dotty/blob/f128063530e9e0c22a761470a79103ef5b823dbe/compiler/src/dotty/tools/dotc/typer/Namer.scala#L168)

### PR DESCRIPTION
**Pull request by susliko**
Fixes #263 and #342
`_if_condition` extracted from `if_expression` and given a magic dynamic
precedence of 4

### CODE REVIEW

File: test/highlight/basics.scala [hunk 1]
```diff
@@ -26,19 +26,19 @@ object Hello {
   trait Test {
 // ^ keyword 
 //       ^ type
-     def meth(i: Int)(implicit x: Boolean) = ???
+    def meth(i: Int)(implicit x: Boolean) = ???
```
Comment (susliko) [inline test/highlight/basics.scala:29]: I've made definitions in this trait to be of the same indentation as
```scala
trait Foo {
    def meth = ???
  def anonFun = ???
}
```
is parsed with an ERROR and this causes highlighting test to flack

We might want to open an issue on this, but I'm reluctant to doing this as `scalac` raises a warning for this code
```
-- Warning: foo.scala:3:2 ------------------------------------------------------
3 |  def anonFun = ???
  |  ^
  |  Line is indented too far to the left, or a `}` is missing
```

File: grammar.js [hunk 2]
```diff
@@ -1104,22 +1106,25 @@ module.exports = grammar({
...
+        ),
+        field("consequence", $._indentable_expression),
+        optional(seq("else", field("alternative", $._indentable_expression))),
+      )),
+
+    // NOTE(susliko): _if_condition and its magic dynamic precedence were introduced as a fix to
+    // https://github.com/tree-sitter/tree-sitter-scala/issues/263 and
+    // https://github.com/tree-sitter/tree-sitter-scala/issues/342 
+    // Neither do I understand why this works, nor have I found a better solution
+    _if_condition: $ => prec.dynamic(4, choice(
```
Comment (susliko) [inline grammar.js:1124]: I'm ashamed of this

File: corpus/expressions.txt [hunk 3]
```diff
@@ -404,6 +404,11 @@ class C:
     else
       ()
 
+    if (a) || b(c) then return true
```
Comment (eed3si9n) [inline corpus/expressions.txt:407]: Nice!

**Review by eed3si9n — APPROVED** [review]
Thanks @susliko!
I've tried this one in the past and gave up, so happy to see this fix.

**Pull request merged**